services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/map/ProcessedMapRepositoryImpl.java

```diff
@@ -0,0 +1,289 @@
+        }
+
+        if (compact){
+            query.fields().exclude("recordGeneratedAt", "properties.validationMessages");
+        }else{
+            query.fields().exclude("recordGeneratedAt");
+        }
+
+        query.addCriteria(Criteria.where("properties.timeStamp").gte(startTimeString).lte(endTimeString));
+        return query;
+    }
+
+    public long getQueryResultCount(Query query) {
+        return mongoTemplate.count(query, ProcessedMap.class, collectionName);
+    }
+
+    public long getQueryFullCount(Query query){
+        int limit = query.getLimit();
+        query.limit(-1);
+        long count = mongoTemplate.count(query, ProcessedMap.class, collectionName);
+        query.limit(limit);
+        return count;
+    }
+
+    public List<ProcessedMap<LineString>> findProcessedMaps(Query query) {
+        List<Map> documents = mongoTemplate.find(query, Map.class, collectionName);
+        List<ProcessedMap<LineString>> convertedList = new ArrayList<>();
+        for (Map document : documents) {
+            document.remove("_id");
+            ProcessedMap<LineString> bsm = mapper.convertValue(document, processedMapTypeReference);
+            convertedList.add(bsm);
+        }
+        return convertedList;
+    }
+
+    public List<IntersectionReferenceData> getIntersectionIDs() {
+        MongoCollection<Document> collection = mongoTemplate.getCollection(collectionName);
+        DistinctIterable<Integer> docs = collection.distinct("properties.intersectionId", Integer.class);
+        List<IntersectionReferenceData> referenceDataList;
+        try (MongoCursor<Integer> results = docs.iterator()) {
+            referenceDataList = new ArrayList<>();
+            while (results.hasNext()) {
+                Integer intersectionId = results.next();
+                Bson projectionFields = Projections.fields(
+                        Projections.include("properties.intersectionId", "properties.originIp",
+                                "properties.refPoint.latitude", "properties.refPoint.longitude", "properties.intersectionName"),
+                        Projections.excludeId());
+                try {
+                    Document document = collection.find(eq("properties.intersectionId", intersectionId))
+                            .projection(projectionFields).sort(Sorts.descending("properties.timeStamp")).maxTime(props.getMongoTimeoutMs(), TimeUnit.MILLISECONDS).first();
+                    if (document != null) {
+                        IntersectionReferenceData data = new IntersectionReferenceData();
+                        Document properties = document.get("properties", Document.class);
+                        if (properties != null) {
+                            Document refPoint = properties.get("refPoint", Document.class);
+                            data.setIntersectionID(intersectionId);
+                            data.setRoadRegulatorID("-1");
```

Comment:
question: is this just a setting we're not using for the CV Manager application? Should we leave it unset instead?

Reply:
This is currently a required field, even though it is unused. I have created a task to remove road regulator ID once and for all: https://dev.azure.com/SOC-OIT/CDOT/_backlogs/backlog/CDOT%20Connected%20Vehicles/Backlog%20items/?workitem=289254

Reply:
Shouldn't we allow for the possibility that some users might want to use CV Manager in situations where there are multiple regions (Road Regulator IDs or Road Authority IDs: one or the other is technically required per the 2024 spec), and that some instrumentation could be configured to send them?  (Even though I know we have yet to see one in the wild..)

Reply:
Discussed this with our team, and think that removing the Road Regulator ID from the UI makes sense - it may be kept in the conflictmonitor objects, but not queried on by the UI or API. The work item has been updated accordingly